File: DailyChallenges/Scala/LeetCode_451.scala
object Solution {
    def frequencySort(s: String): String = {
        val m = s.groupMapReduce(identity)(_=>1)(_+_)
        s.sortWith((c1,c2)=> { if (m(c1)==m(c2)) c1>c2 else m(c1)>m(c2)})
    }
}

File: DailyChallenges/CPP/LeetCode_451.cpp
// 451. Sort Characters By Frequency @ 2022/12/03

class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> m;
        for (auto ch:s)
            ++m[ch];
        sort(s.begin(), s.end(), [&m](const char& c1, const char& c2){ return m[c1]==m[c2] ? c1 >c2 : m[c1]>m[c2]; });
        return s;
    }
};
